pennylane/templates/subroutines/trotter.py

```diff
@@ -490,3 +492,335 @@ def compute_decomposition(*args, **kwargs):
+
+    .. math::
+
+        \begin{align}
+            S_{1}(t) &= \Pi_{j=0}^{N} \ e^{i t O_{j}} \\
+            S_{2}(t) &= \Pi_{j=0}^{N} \ e^{i \frac{t}{2} O_{j}} \cdot \Pi_{j=N}^{0} \ e^{i \frac{t}{2} O_{j}} \\
+            &\vdots \\
+            S_{m}(t) &= S_{m-2}(p_{m}t)^{2} \cdot S_{m-2}((1-4p_{m})t) \cdot S_{m-2}(p_{m}t)^{2},
+        \end{align}
+
+    where the coefficient is :math:`p_{m} = 1 / (4 - \sqrt[m - 1]{4})`. The :math:`m`th order,
+    :math:`n`-step Suzuki-Trotter approximation is then defined as:
+
+    .. math:: e^{iHt} \approx \left [S_{m}(t / n)  \right ]^{n}.
+
+    For more details see `J. Math. Phys. 32, 400 (1991) <https://pubs.aip.org/aip/jmp/article-abstract/32/2/400/229229>`_.
+
+    Suppose we have direct access to the operators which represent the exponentiated terms of
+    a Hamiltonian:
+
+    .. math:: \{ \hat{U}_{j} = e^{i t O_{j}} for j \in [1, N] \}.
+
+    Given a quantum circuit which uses these :math:`\hat{U}_{j}` operators to represents the
+    first order expansion :math:`S_{1}(t)`, this function expands it to any higher order Suzuki-Trotter product.
+
+    .. warning::
+
+        :code:`trotterize()` requires the :code:`qfunc` argument to be a function with a specific call
+        signature. The first argument of the :code:`qfunc` function should be a time parameter which will be modified according to the
+        Suzuki-Trotter product formula. The wires required by the circuit should be either the last
+        positional argument or the first keyword argument:
+        :code:`qfunc((time, arg1, ..., arg_n, wires=[...], kwarg_1, ..., kwarg_n))`
+
+    Args:
+        qfunc (Callable): the first-order expansion given as a callable function which queues operations
+        n (int): an integer representing the number of Trotter steps to perform
+        order (int): an integer (:math:`m`) representing the order of the approximation (must be 1 or even)
+        reverse (bool): if true, reverse the order of the operations queued by :code:`qfunc`
+        name (str): an optional name for the instance
+
+    Returns:
+        Callable: a function with the same signature as :code:`qfunc`, when called it queues an instance of
+            :class:`~.TrotterizedQfunc`
+
+    **Example**
+
+    .. code-block:: python3
+
+        def first_order_expansion(time, theta, phi, wires, flip=False):
+            "This is the first order expansion (U_1)."
+            qml.RX(time*theta, wires[0])
+            qml.RY(time*phi, wires[1])
+            if flip:
+                qml.CNOT(wires=wires[:2])
+
+        @qml.qnode(qml.device("default.qubit"))
+        def my_circuit(time, theta, phi, num_trotter_steps):
+            qml.trotterize(
+                first_order_expansion,
+                n=num_trotter_steps,
+                order=2,
+            )(time, theta, phi, wires=['a', 'b', 'c'], flip=True)
+            return qml.state()
```

Comment:
Is there a reason one of these has 3 wires and the other has 2 wires?
```python
print(qml.draw(my_circuit, level=2)(1,1,1,1), '\n')
print(qml.draw(my_circuit, level=3)(1,1,1,1))
```
```
a: ─╭TrotterizedQfunc(1.00,1.00,1.00)─┤  State
b: ─├TrotterizedQfunc(1.00,1.00,1.00)─┤  State
c: ─╰TrotterizedQfunc(1.00,1.00,1.00)─┤  State 

a: ──RX(0.50)─╭●─╭●──RX(0.50)─┤  State
b: ──RY(0.50)─╰X─╰X──RY(0.50)─┤  State
```

Reply:
Ya it's more of a consequence of `qml.draw()`. In the example we gave our operation 3 wires, but in the decomposition there are only 2, so when we draw it after decomposition the extra wire does not get displayed.

Reply:
I can change the example to only use 2 wires if it's confusing.

Reply:
Fixed 👍🏼